NonPersistentTopics - [myuser] Failed to get list of topics under namespace public/default

**Describe the bug**
Unable to list the non-persistent topics using the command line, what other permission is required for the role?

My subject is  `myuser` and authorized by the below command using superuser role:
```bash
$ pulsar-admin namespaces grant-permission public/default --role myuser --actions produce,consume,sources,sinks,functions,packages
```
 
Topics list command:
```bash
$ pulsar-admin --admin-url https://dev.pulsar.xyz.com:8081 --auth-plugin "org.apache.pulsar.client.impl.auth.AuthenticationToken" --auth-params "token:<token>" topics list public/default
```

Error:
```
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: 2022-02-09T12:38:16,128+0000 [AsyncHttpClient-48-1] ERROR org.apache.pulsar.broker.admin.v2.NonPersistentTopics - [myuser] Failed to get list of topics under namespace public/default
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: java.util.concurrent.ExecutionException: org.apache.pulsar.client.admin.PulsarAdminException$NotAuthorizedException: Unauthorized to validateNamespaceOperation for operation [GET_BUNDLE] on namespace [public/default]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357) ~[?:1.8.0_312]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1908) ~[?:1.8.0_312]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.apache.pulsar.broker.admin.v2.NonPersistentTopics.lambda$getList$4(NonPersistentTopics.java:410) ~[org.apache.pulsar-pulsar-broker-2.9.1.jar:2.9.1]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at java.util.concurrent.CompletableFuture.uniHandle(CompletableFuture.java:836) ~[?:1.8.0_312]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at java.util.concurrent.CompletableFuture$UniHandle.tryFire(CompletableFuture.java:811) ~[?:1.8.0_312]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488) ~[?:1.8.0_312]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:1990) ~[?:1.8.0_312]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.apache.pulsar.client.admin.internal.TopicsImpl$5.failed(TopicsImpl.java:287) ~[org.apache.pulsar-pulsar-client-admin-original-2.9.1.jar:2.9.1]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.glassfish.jersey.client.JerseyInvocation$1.failed(JerseyInvocation.java:882) ~[org.glassfish.jersey.core-jersey-client-2.34.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.glassfish.jersey.client.JerseyInvocation$1.completed(JerseyInvocation.java:863) ~[org.glassfish.jersey.core-jersey-client-2.34.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.glassfish.jersey.client.ClientRuntime.processResponse(ClientRuntime.java:229) ~[org.glassfish.jersey.core-jersey-client-2.34.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.glassfish.jersey.client.ClientRuntime.access$200(ClientRuntime.java:62) ~[org.glassfish.jersey.core-jersey-client-2.34.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.glassfish.jersey.client.ClientRuntime$2.lambda$response$0(ClientRuntime.java:173) ~[org.glassfish.jersey.core-jersey-client-2.34.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248) ~[org.glassfish.jersey.core-jersey-common-2.34.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244) ~[org.glassfish.jersey.core-jersey-common-2.34.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.glassfish.jersey.internal.Errors.process(Errors.java:292) ~[org.glassfish.jersey.core-jersey-common-2.34.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.glassfish.jersey.internal.Errors.process(Errors.java:274) ~[org.glassfish.jersey.core-jersey-common-2.34.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.glassfish.jersey.internal.Errors.process(Errors.java:244) ~[org.glassfish.jersey.core-jersey-common-2.34.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:288) ~[org.glassfish.jersey.core-jersey-common-2.34.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.glassfish.jersey.client.ClientRuntime$2.response(ClientRuntime.java:173) ~[org.glassfish.jersey.core-jersey-client-2.34.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.apache.pulsar.client.admin.internal.http.AsyncHttpConnector.lambda$apply$1(AsyncHttpConnector.java:228)~[org.apache.pulsar-pulsar-client-admin-original-2.9.1.jar:2.9.1]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:774) ~[?:1.8.0_312]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:750) ~[?:1.8.0_312]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488) ~[?:1.8.0_312]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:1975) ~[?:1.8.0_312]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.apache.pulsar.client.admin.internal.http.AsyncHttpConnector.lambda$retryOperation$4(AsyncHttpConnector.java:270) ~[org.apache.pulsar-pulsar-client-admin-original-2.9.1.jar:2.9.1]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:774) [?:1.8.0_312]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:750) [?:1.8.0_312]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:488) [?:1.8.0_312]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:1975) [?:1.8.0_312]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.asynchttpclient.netty.NettyResponseFuture.loadContent(NettyResponseFuture.java:222) [org.asynchttpclient-async-http-client-2.12.1.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.asynchttpclient.netty.NettyResponseFuture.done(NettyResponseFuture.java:257) [org.asynchttpclient-async-http-client-2.12.1.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.asynchttpclient.netty.handler.AsyncHttpClientHandler.finishUpdate(AsyncHttpClientHandler.java:241) [org.asynchttpclient-async-http-client-2.12.1.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.asynchttpclient.netty.handler.HttpHandler.handleChunk(HttpHandler.java:114) [org.asynchttpclient-async-http-client-2.12.1.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.asynchttpclient.netty.handler.HttpHandler.handleRead(HttpHandler.java:143) [org.asynchttpclient-async-http-client-2.12.1.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.asynchttpclient.netty.handler.AsyncHttpClientHandler.channelRead(AsyncHttpClientHandler.java:78) [org.asynchttpclient-async-http-client-2.12.1.jar:?]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103) [io.netty-netty-codec-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324) [io.netty-netty-codec-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:296) [io.netty-netty-codec-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:658) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:584) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496) [io.netty-netty-transport-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986) [io.netty-netty-common-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [io.netty-netty-common-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [io.netty-netty-common-4.1.72.Final.jar:4.1.72.Final]
Feb 09 12:38:16 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at java.lang.Thread.run(Thread.java:748) [?:1.8.0_312]
```

`proxy.conf`
```apache
superUserRoles=
authorizationEnabled=true
authorizationProvider=org.apache.pulsar.broker.authorization.PulsarAuthorizationProvider
authenticationEnabled=true
authenticationProviders=org.apache.pulsar.broker.authentication.AuthenticationProviderToken
anonymousUserRole=
brokerClientAuthenticationPlugin=org.apache.pulsar.client.impl.auth.AuthenticationToken
brokerClientAuthenticationParameters=file:///my/token/path.txt
```

`broker.conf`
```apache
authenticationEnabled=true
authenticationProviders=org.apache.pulsar.broker.authentication.AuthenticationProviderToken
authorizationEnabled=true
authorizationProvider=org.apache.pulsar.broker.authorization.PulsarAuthorizationProvider

superUserRoles=mysuperadmin

brokerClientTlsEnabled=false
brokerClientAuthenticationPlugin=org.apache.pulsar.client.impl.auth.AuthenticationToken
brokerClientAuthenticationParameters=file:///my/token/path.txt
brokerClientTrustCertsFilePath=
```

**Expected behavior**
Should list the topics

### Using Apache Pulsar 2.9.1

**Desktop (please complete the following information):**
```
NAME="Red Hat Enterprise Linux Server"
VERSION="7.7 (Maipo)"
ID="rhel"
ID_LIKE="fedora"
VARIANT="Server"
VARIANT_ID="server"
VERSION_ID="7.7"
PRETTY_NAME="Red Hat Enterprise Linux Server 7.7 (Maipo)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:7.7:GA:server"
HOME_URL="https://www.redhat.com/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"

REDHAT_BUGZILLA_PRODUCT="Red Hat Enterprise Linux 7"
REDHAT_BUGZILLA_PRODUCT_VERSION=7.7
REDHAT_SUPPORT_PRODUCT="Red Hat Enterprise Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="7.7"
```

I have retested from the beginning, create tenant with admin roles then create a namespace and then using admin role create token and called pulsar admin to list the topic of the new tenant and namespace but still same error, admin client shows the below error:
```
HTTP 500 Server Error

Reason: HTTP 500 Server Error
```

Steps:
```bash
$ pulsar-admin --admin-url https://dev.pulsar.xyz.com:8081 --auth-plugin "org.apache.pulsar.client.impl.auth.AuthenticationToken" --auth-params "token:<token>" tenants create mytenant --admin-roles my-admin-role
$ pulsar-admin --admin-url https://dev.pulsar.xyz.com:8081 --auth-plugin "org.apache.pulsar.client.impl.auth.AuthenticationToken" --auth-params "token:<token>" namespaces create mytenant/np1 
$ pulsar tokens create --private-key file:///path/to/private/key --subject my-admin-role --expiry-time 60m
$ pulsar-admin --admin-url https://dev.pulsar.xyz.com:8081 --auth-plugin "org.apache.pulsar.client.impl.auth.AuthenticationToken" --auth-params "token:<token>" topics list mytenant/np1 
HTTP 500 Server Error

Reason: HTTP 500 Server Error
```

Pulsar broker log:
```
Caused by: org.apache.pulsar.client.admin.PulsarAdminException$NotAuthorizedException: Unauthorized to validateNamespaceOperation for operation [GET_BUNDLE] on namespace [mytenant/np1]
Feb 09 13:36:47 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.apache.pulsar.client.admin.internal.BaseResource.getApiException(BaseResource.java:228) ~[org.apache.pulsar-pulsar-client-admin-original-2.9.1.jar:2.9.1]
Feb 09 13:36:47 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: ... 56 more
Feb 09 13:36:47 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: Caused by: javax.ws.rs.ForbiddenException: HTTP 403 Forbidden
Feb 09 13:36:47 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.glassfish.jersey.client.JerseyInvocation.convertToException(JerseyInvocation.java:945) ~[org.glassfish.jersey.core-jersey-client-2.34.jar:?]
Feb 09 13:36:47 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: at org.glassfish.jersey.client.JerseyInvocation.access$700(JerseyInvocation.java:82) ~[org.glassfish.jersey.core-jersey-client-2.34.jar:?]
Feb 09 13:36:47 ip-10-11-2-26.eu-north-1.compute.internal pulsar[31737]: ... 54 more
```

--- a/pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/NonPersistentTopics.java
+++ b/pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v1/NonPersistentTopics.java
@@ -200,6 +200,7 @@ public class NonPersistentTopics extends PersistentTopics {
         Policies policies = null;
         NamespaceName nsName = null;
         try {
+            validateNamespaceName(property, cluster, namespace);
             validateNamespaceOperation(namespaceName, NamespaceOperation.GET_TOPICS);
             policies = getNamespacePolicies(property, cluster, namespace);
             nsName = NamespaceName.get(property, cluster, namespace);
@@ -237,22 +238,19 @@ public class NonPersistentTopics extends PersistentTopics {
             }
         }
 
-        final List<String> topics = Lists.newArrayList();
-        FutureUtil.waitForAll(futures).handle((result, exception) -> {
-            for (int i = 0; i < futures.size(); i++) {
-                try {
-                    if (futures.get(i).isDone() && futures.get(i).get() != null) {
-                        topics.addAll(futures.get(i).get());
+        FutureUtil.waitForAll(futures).whenComplete((result, ex) -> {
+            if (ex != null) {
+                resumeAsyncResponseExceptionally(asyncResponse, ex);
+            } else {
+                final List<String> topics = Lists.newArrayList();
+                for (int i = 0; i < futures.size(); i++) {
+                    List<String> topicList = futures.get(i).join();
+                    if (topicList != null) {
+                        topics.addAll(topicList);
                     }
-                } catch (InterruptedException | ExecutionException e) {
-                    log.error("[{}] Failed to get list of topics under namespace {}/{}/{}", clientAppId(), property,
-                            cluster, namespace, e);
-                    asyncResponse.resume(new RestException(e instanceof ExecutionException ? e.getCause() : e));
-                    return null;
                 }
+                asyncResponse.resume(topics);
             }
-            asyncResponse.resume(topics);
-            return null;
         });
     }
 
@@ -269,6 +267,7 @@ public class NonPersistentTopics extends PersistentTopics {
                                           @PathParam("bundle") String bundleRange) {
         log.info("[{}] list of topics on namespace bundle {}/{}/{}/{}", clientAppId(), property, cluster, namespace,
                 bundleRange);
+        validateNamespaceName(property, cluster, namespace);
         validateNamespaceOperation(namespaceName, NamespaceOperation.GET_BUNDLE);
         Policies policies = getNamespacePolicies(property, cluster, namespace);
         if (!cluster.equals(Constants.GLOBAL_CLUSTER)) {

--- a/pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/NonPersistentTopics.java
+++ b/pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/NonPersistentTopics.java
@@ -415,25 +415,19 @@ public class NonPersistentTopics extends PersistentTopics {
         }
 
         final List<String> topics = Lists.newArrayList();
-        FutureUtil.waitForAll(futures).handle((result, exception) -> {
-            for (int i = 0; i < futures.size(); i++) {
-                try {
-                    if (futures.get(i).isDone() && futures.get(i).get() != null) {
-                        topics.addAll(futures.get(i).get());
-                    }
-                } catch (InterruptedException | ExecutionException e) {
-                    log.error("[{}] Failed to get list of topics under namespace {}", clientAppId(), namespaceName, e);
-                    asyncResponse.resume(new RestException(e instanceof ExecutionException ? e.getCause() : e));
-                    return null;
+        FutureUtil.waitForAll(futures).whenComplete((result, ex) -> {
+            if (ex != null) {
+                resumeAsyncResponseExceptionally(asyncResponse, ex);
+            } else {
+                for (int i = 0; i < futures.size(); i++) {
+                    topics.addAll(futures.get(i).join());
                 }
+                final List<String> nonPersistentTopics =
+                        topics.stream()
+                                .filter(name -> !TopicName.get(name).isPersistent())
+                                .collect(Collectors.toList());
+                asyncResponse.resume(nonPersistentTopics);
             }
-
-            final List<String> nonPersistentTopics =
-                    topics.stream()
-                            .filter(name -> !TopicName.get(name).isPersistent())
-                            .collect(Collectors.toList());
-            asyncResponse.resume(nonPersistentTopics);
-            return null;
         });
     }
 

--- a/pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/NonPersistentTopics.java
+++ b/pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/NonPersistentTopics.java
@@ -414,13 +414,16 @@ public class NonPersistentTopics extends PersistentTopics {
             }
         }
 
-        final List<String> topics = Lists.newArrayList();
         FutureUtil.waitForAll(futures).whenComplete((result, ex) -> {
             if (ex != null) {
                 resumeAsyncResponseExceptionally(asyncResponse, ex);
             } else {
+                final List<String> topics = Lists.newArrayList();
                 for (int i = 0; i < futures.size(); i++) {
-                    topics.addAll(futures.get(i).join());
+                    List<String> topicList = futures.get(i).join();
+                    if (topicList != null) {
+                        topics.addAll(topicList);
+                    }
                 }
                 final List<String> nonPersistentTopics =
                         topics.stream()

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/auth/AuthorizationTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/auth/AuthorizationTest.java
@@ -18,14 +18,17 @@
  */
 package org.apache.pulsar.broker.auth;
 
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.assertFalse;
 import static org.testng.Assert.assertTrue;
 import static org.testng.Assert.fail;
-
 import java.util.EnumSet;
-
 import org.apache.pulsar.broker.authorization.AuthorizationService;
+import org.apache.pulsar.client.admin.PulsarAdmin;
 import org.apache.pulsar.client.admin.PulsarAdminBuilder;
+import org.apache.pulsar.client.admin.PulsarAdminException;
+import org.apache.pulsar.common.naming.TopicDomain;
 import org.apache.pulsar.common.naming.TopicName;
 import org.apache.pulsar.common.policies.data.AuthAction;
 import org.apache.pulsar.common.policies.data.ClusterData;
@@ -35,7 +38,6 @@ import org.testng.Assert;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
-
 import com.google.common.collect.Sets;
 
 @Test(groups = "flaky")
@@ -227,6 +229,28 @@ public class AuthorizationTest extends MockedPulsarServiceBaseTest {
         admin.namespaces().deleteNamespace("p1/c1/ns1");
         admin.tenants().deleteTenant("p1");
         admin.clusters().deleteCluster("c1");
+    }
+
+    @Test
+    public void testGetListWithoutGetBundleOp() throws Exception {
+        String tenant = "p1";
+        String namespace = "p1/ns1";
+        admin.clusters().createCluster("c1", ClusterData.builder().build());
+        admin.tenants().createTenant(tenant, new TenantInfoImpl(Sets.newHashSet("role1"), Sets.newHashSet("c1")));
+        admin.namespaces().createNamespace(namespace);
+        admin.namespaces().grantPermissionOnNamespace(namespace, "pass.pass2", EnumSet.of(AuthAction.produce));
+        PulsarAdmin admin2 = PulsarAdmin.builder().serviceHttpUrl(brokerUrl != null
+                        ? brokerUrl.toString()
+                        : brokerUrlTls.toString())
+                .authentication(new MockAuthentication("pass.pass2"))
+                .build();
+        when(pulsar.getAdminClient()).thenReturn(admin2);
+        try {
+            admin2.topics().getList(namespace, TopicDomain.non_persistent);
+        } catch (Exception ex) {
+            assertTrue(ex instanceof PulsarAdminException.NotAuthorizedException);
+            assertEquals(ex.getMessage(), "Unauthorized to validateNamespaceOperation for operation [GET_BUNDLE] on namespace [p1/ns1]");
+        }
     }
 
     private static void waitForChange() {

--- a/pulsar-broker/src/test/java/org/apache/pulsar/broker/auth/AuthorizationTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/broker/auth/AuthorizationTest.java
@@ -234,11 +234,14 @@ public class AuthorizationTest extends MockedPulsarServiceBaseTest {
     @Test
     public void testGetListWithoutGetBundleOp() throws Exception {
         String tenant = "p1";
-        String namespace = "p1/ns1";
+        String namespaceV1 = "p1/global/ns1";
+        String namespaceV2 = "p1/ns2";
         admin.clusters().createCluster("c1", ClusterData.builder().build());
         admin.tenants().createTenant(tenant, new TenantInfoImpl(Sets.newHashSet("role1"), Sets.newHashSet("c1")));
-        admin.namespaces().createNamespace(namespace);
-        admin.namespaces().grantPermissionOnNamespace(namespace, "pass.pass2", EnumSet.of(AuthAction.produce));
+        admin.namespaces().createNamespace(namespaceV1, Sets.newHashSet("c1"));
+        admin.namespaces().grantPermissionOnNamespace(namespaceV1, "pass.pass2", EnumSet.of(AuthAction.produce));
+        admin.namespaces().createNamespace(namespaceV2, Sets.newHashSet("c1"));
+        admin.namespaces().grantPermissionOnNamespace(namespaceV2, "pass.pass2", EnumSet.of(AuthAction.produce));
         PulsarAdmin admin2 = PulsarAdmin.builder().serviceHttpUrl(brokerUrl != null
                         ? brokerUrl.toString()
                         : brokerUrlTls.toString())
@@ -246,10 +249,16 @@ public class AuthorizationTest extends MockedPulsarServiceBaseTest {
                 .build();
         when(pulsar.getAdminClient()).thenReturn(admin2);
         try {
-            admin2.topics().getList(namespace, TopicDomain.non_persistent);
+            admin2.topics().getList(namespaceV1, TopicDomain.non_persistent);
         } catch (Exception ex) {
             assertTrue(ex instanceof PulsarAdminException.NotAuthorizedException);
-            assertEquals(ex.getMessage(), "Unauthorized to validateNamespaceOperation for operation [GET_BUNDLE] on namespace [p1/ns1]");
+            assertEquals(ex.getMessage(), "Unauthorized to validateNamespaceOperation for operation [GET_BUNDLE] on namespace [p1/global/ns1]");
+        }
+        try {
+            admin2.topics().getList(namespaceV2, TopicDomain.non_persistent);
+        } catch (Exception ex) {
+            assertTrue(ex instanceof PulsarAdminException.NotAuthorizedException);
+            assertEquals(ex.getMessage(), "Unauthorized to validateNamespaceOperation for operation [GET_BUNDLE] on namespace [p1/ns2]");
         }
     }
 

